gnovm/pkg/gnolang/values_string.go

```diff
@@ -7,6 +7,43 @@ import (
 	"strings"
 )
 
+// This indicates the maximum ancticipated depth of the stack when printing a Value type.
+const defaultSeenValuesSize = 32
+
+type seenValues struct {
+	values []Value
+}
+
+func (sv *seenValues) Put(v Value) {
+	sv.values = append(sv.values, v)
+}
+
+func (sv *seenValues) Contains(v Value) bool {
+	for _, vv := range sv.values {
+		if vv == v {
+			return true
+		}
```

Comment:
okay for now, though I'd switch to slices.Contains once we switch to 1.21 as the min version